docs/how_to_dev_frontend_en.md

```diff
@@ -0,0 +1,105 @@
+# VisualDL Web App Develop
+
+VisualDL has two major components. One is the Python/C++ SDK that logs the matrix during training,
+the other is the web-app that display visualized training data.
+
+The web-app reads the extract data and draw scalars graph or histograms graphs.
+The web-app is where VisualDL transforms the code data into graphs so users can visualize the training progress and make a better network design evaluation.
+
+This article will go over the basic web-app architecture and development steps.
+
+## Web App Overview
+The VisualDL web-app home page is constructed by three major Javascript files
+1. manifest.js: The manifest configures basic settings  
+1. vendor.js: The vendor.js stores all the 3rd party libraries.
+1. index.js: The index.js contains all the graph algorithms. All navigations, setting panels are implemented here.
+It finds the 'root' DOM in the index.html and attaches the web-app to it.
+
+The graph algorithms are defined in the `.san` files. san is the ECharts' graph framework. It can pilot charts and graphs with just a few lines of code.
+To learn more about ECharts's framework, please visit [ECharts](https://ecomfe.github.io/echarts-doc/public/en/index.html).
+
+## npm package
+The VisualDL uses npm to manage the JavaScript files. npm is the package manager for JavaScript and is a great tool to manage multiple dependencies. To confirm that you have npm installed. Please run this command
+``` bash
+npm -v
+```
+If your environment does not have npm already, please [download npm](https://www.npmjs.com/get-npm).
+
+The basic packaging information is stored in `package.json`. It specifies build scripts and dependencies for each environment.
+npm reads `package.json` to build different package according to the packaging script.
+
+Here is a snippet of the package.json
+```
+{
+  "name": "VisualDL",
+  "version": "1.0.0",
+  "description": "Visualization toolkit for deep learning",
+  "scripts": {
+    "release": "cross-env NODE_ENV=production node ./tool/build.js",
+    "build": "cross-env NODE_ENV=dev node ./tool/build.js",
+    "dev": "cross-env NODE_ENV=dev node tool/dev-server.js",
+    "lint": "./node_modules/fecs/bin/fecs --rule"
+  },
+  "engines": {
+    "node": ">= 6.4.0"
+  },
+  "dependencies": {
+    "axios": "^0.16.1",
+    "csshint": "^0.3.3"
+    ...
+    ...
+    ...
+  }
+}
+```
+
+
+## Development Flow
+The VisualDL is an open source project. Everyone is welcome to contribute to the project. To get start the process, please refer to [Contribute Code](contribute_code_en.md) to set up the environment.
+
+1. Create Development Server
```

Comment:
More than creating a development server, you are definitely `Running the development server`.

Since it is not common to run a dev server for front-end separately from the server-side, you might have to also explicitly explain that as a part of the architecture.

Reply:
Ah, I see. I will explicitly explain that we are only running the frontend development server.